kong/plugins/grpc-web/handler.lua

```diff
@@ -0,0 +1,97 @@
+-- Copyright (c) Kong Inc. 2020
+
+local deco = require "kong.plugins.grpc-web.deco"
+
+local ngx = ngx
+local kong = kong
+
+local string_format = string.format
+
+local ngx_arg = ngx.arg
+local ngx_var = ngx.var
+
+local kong_request_get_path = kong.request.get_path
+local kong_request_get_header = kong.request.get_header
+local kong_request_get_method = kong.request.get_method
+local kong_request_get_raw_body = kong.request.get_raw_body
+local kong_response_exit = kong.response.exit
+local kong_response_set_header = kong.response.set_header
+local kong_service_request_set_header = kong.service.request.set_header
+local kong_service_request_set_raw_body = kong.service.request.set_raw_body
+
+
+local grpc_web = {
+  PRIORITY = 3,
+  VERSION = '0.1.0',
+}
+
+
+local CORS_HEADERS = {
+  ["Content-Type"] = "application/grpc-web-text+proto",
+  ["Access-Control-Allow-Origin"] = "*",
+  ["Access-Control-Allow-Methods"] = "POST",
+  ["Access-Control-Allow-Headers"] = "content-type,x-grpc-web,x-user-agent",
+}
+
+function grpc_web:access(conf)
+  kong_response_set_header("Access-Control-Allow-Origin", "*")
+
+  if kong_request_get_method() == "OPTIONS" then
+    return kong_response_exit(200, "OK", CORS_HEADERS)
+  end
+
+
+  local dec, err = deco.new(
+    kong_request_get_header("Content-Type"),
+    kong_request_get_path(), conf.proto)
+
+  if not dec then
+    kong.log.err(err)
+    return kong_response_exit(500, err)
```

Comment:
Shouldn't the error status be `400` here?

It seems the only thing that can make `deco.new` fail is a missing proto file.

Reply:
my original intention was that a misconfiguration should be "server error" (since there's no "proxy error" family), but i realize that now a request to an nonexistent method can also fail, so yes, 400 would be better.